data/src/main/java/com/example/data/db/dao/TableSizeDeleteCityDao.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.data.db.dao
+
+import androidx.room.Dao
+import androidx.room.Query
+import androidx.room.Transaction
+
+@Dao
+abstract class TableSizeDeleteCityDao {
+
+    @Query("SELECT COUNT(id) FROM last_chosen_cities_entity")
+    abstract fun getTableSize(): Int
+
+    @Query("DELETE FROM last_chosen_cities_entity WHERE id = (SELECT id FROM last_chosen_cities_entity ORDER BY id LIMIT 1)")
+    abstract fun deleteExtraCity()
+
+    @Transaction
+    open  fun getTableSizeAndDeleteExtraCity(){
```

Comment:
This transaction doesn't make sense, because method is not called anywhere.

Reply:
Fixed